### PR TITLE
Feature/ds max anno

### PR DESCRIPTION
**Pull request by jemten**
### This PR adds | fixes:

- Adds a check of the vcfanno toml to see if the annotation keys are present in the reference vcf
- Adds the capability to process a preops key in the vcfanno toml config and from that generate a new vcfanno config. That config is used to annotate the reference file using vcfanno. 

### How to test:

- Automatic and continuous test suite

### Expected outcome:
- [ ] Installation, unit and integration test suite pass

### Review:
- [ ] Code review
- [ ] Tests pass

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes. If applicable record manual test results in PR header
- [x] **MINOR** - when you add functionality in a backwards compatible manner. If applicable record manual test results in PR header
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: t/gnu_split.t [hunk 1]
```diff
@@ -1,136 +1,125 @@
...
+      . $MIP::Program::Gnu::Coreutils::VERSION
+      . $COMMA
+      . $SPACE . q{Perl}
+      . $SPACE
+      . $PERL_VERSION
+      . $SPACE
+      . $EXECUTABLE_NAME );
 
-diag("Test gnu_split $MIP::Program::Gnu::Coreutils::VERSION, Perl $^V, $EXECUTABLE_NAME");
+Readonly my $FILES => 12;
```
Comment (henrikstranneheim) [inline t/gnu_split.t:62]:
```suggestion
Readonly my $NUMBER_OF_FILES => 12;
```

File: t/gnu_split.t [hunk 2]
```diff
@@ -1,136 +1,125 @@
 #!/usr/bin/env perl
```
Comment (henrikstranneheim) [inline t/gnu_split.t:1]: NIce clean-up! ⭐

File: lib/MIP/Vcfanno.pm [hunk 3]
```diff
@@ -25,7 +25,97 @@ BEGIN {
...
+            }
+        );
+
+        if ( %preops and any { $_ eq $vcf_id_tag } @{ $preops{names} } ) {
+
+            push @{ $preops_href->{$annotation_file_path}{annotation} }, \%preops;
+            return;
+        }
+
+        push @{ $missing_tag_href->{$annotation_file_path} }, $vcf_id_tag;
```
Comment (jemten) [inline lib/MIP/Vcfanno.pm:116]: refactor to new sub

File: lib/MIP/Language/Perl.pm [hunk 4]
```diff
@@ -476,6 +494,40 @@ sub _get_select_contigs_by_col {
...
+
+    ## Find vcf_key
+    my $vcf_header_line = q?'if($_=~/ID\=? . $id . q?,/) { ?;
+
+    ## Write to stdout
+    $vcf_header_line .= q?print $_} ?;
+
+    ## If header is finished quit
+    $vcf_header_line .= q?if($_=~/#CHROM/) {last}'?;
+
```
Comment (jemten) [inline lib/MIP/Language/Perl.pm:527]: strengthen regexp

File: t/gnu_split.t (hunk 2 above)
Comment (jemten) [reply t/gnu_split.t:1]: thanks!

File: lib/MIP/Recipes/Analysis/Variant_annotation.pm [hunk 5]
```diff
@@ -29,10 +45,348 @@ BEGIN {
...
+            file_name   => $temp_outfile_path,
+            file_suffix => q{.vcf},
+        }
+    );
+    my $upper_file_number = $recipe_resource{core_number} - 1;
+    my @splitted_infiles  = map { $temp_outfile_path_prefix . $UNDERSCORE . $_ }
+      ( '00' .. "$upper_file_number" );
+    my @splitted_outfiles =
+      map { $temp_outfile_path_prefix . $UNDERSCORE . $_ . $DOT . q{vcf.gz} }
+      ( '00' .. "$upper_file_number" );
```
[the remark this answers is not in the document]
Comment (jemten) [reply lib/MIP/Recipes/Analysis/Variant_annotation.pm:213]: the splitted files get a two digit suffix starting at 00 and ending at 35 for a 36 core job. This ensures that the range has a leading zero where appropriate. Should probably be q{00} though, or even a maybe even a constant.

**Pull request merged**